clients/cli/src/wrap.rs

```diff
@@ -0,0 +1,224 @@
+use {
+    crate::{
+        common::{parse_pubkey, parse_token_program, process_transaction},
+        config::Config,
+        output::{format_output, println_display},
+        CommandResult, Error,
+    },
+    clap::Args,
+    serde_derive::{Deserialize, Serialize},
+    serde_with::{serde_as, DisplayFromStr},
+    solana_cli_output::{display::writeln_name_value, QuietDisplay, VerboseDisplay},
+    solana_client::nonblocking::rpc_client::RpcClient,
+    solana_pubkey::Pubkey,
+    solana_remote_wallet::remote_wallet::RemoteWalletManager,
+    solana_signature::Signature,
+    solana_signer::Signer,
+    solana_transaction::Transaction,
+    spl_associated_token_account::get_associated_token_address_with_program_id,
+    spl_token_2022::{extension::PodStateWithExtensions, pod::PodAccount},
+    spl_token_wrap::{get_wrapped_mint_address, get_wrapped_mint_authority, instruction::wrap},
+    std::{
+        fmt::{Display, Formatter},
+        rc::Rc,
+        sync::Arc,
+    },
+};
+
+#[derive(Clone, Debug, Args)]
+pub struct WrapArgs {
+    /// The address of the token program that the unwrapped mint belongs to
+    #[clap(value_parser = parse_token_program)]
+    pub unwrapped_token_program: Pubkey,
+
+    /// The address of the unwrapped token account to wrap from
+    #[clap(value_parser = parse_pubkey)]
+    pub unwrapped_token_account: Pubkey,
+
+    /// The address of the escrow account that will hold the unwrapped tokens
+    #[clap(value_parser = parse_pubkey)]
+    pub escrow_account: Pubkey,
+
+    /// The address of the token program that the wrapped mint should belong to
+    #[clap(value_parser = parse_token_program)]
+    pub wrapped_token_program: Pubkey,
+
+    /// The amount of tokens to wrap
+    #[clap(value_parser)]
+    pub amount: u64,
+
+    /// Path to the signer for the transfer authority if different from
+    /// fee payer
+    #[clap(long = "transfer-authority", value_name = "PATH")]
```

Comment:
nit: same here, let clap figure it out automatically
```suggestion
    #[clap(long, value_name = "PATH")]
```